crates/fuel-core/src/query/contract.rs

```diff
@@ -81,13 +88,13 @@ impl ContractQueryContext<'_> {
         })
     }
 
-    pub fn contract_balances(
+    fn contract_balances(
         &self,
         contract_id: ContractId,
         start_asset: Option<AssetId>,
         direction: IterDirection,
-    ) -> impl Iterator<Item = StorageResult<ContractBalance>> + '_ {
-        self.0
-            .contract_balances(contract_id, start_asset, direction)
+    ) -> BoxedIter<StorageResult<ContractBalance>> {
+        self.contract_balances(contract_id, start_asset, direction)
```

Comment:
You don't need to wrap it `into_boxed` because it is already wrapped=) 
<img width="483" alt="image" src="https://user-images.githubusercontent.com/18346821/220441394-13d15d53-f66e-4acb-a189-9c9b7060276f.png">

Could you review all other places with `into_boxed` to avoid double wrapping, please?=)